### ISSUE TITLE
Fortran function call in const.clust fails

### ISSUE THREAD

**Issue by brendanf** [philippec/fonctions_R_git#1]
I've tried to install const.clust from source in R 3.2.3 and 3.0.3.  In both cases, using the function "constrained.clust" yields the following error:

`Error in .Fortran("hcass2", n = as.integer(n), ia = as.integer(hcl$ia),  : 
  "hcass2" not resolved from current namespace (const.clust)`

Edit: I've tried the installation in linux and Windows.

**Comment by philippec** [comment]
You need to build with `R CMD build const.clust`, which will build the module. `hcass2` is a Fortran subroutine for which the code is provided: https://github.com/philippec/fonctions_R_git/blob/master/const.clust/src/hcass2.f. You will need a Fortran compiler like g77.

**Comment by brendanf** [comment]
Thanks for the advice.  However, I have been having no trouble getting the Fortran compiled; this happens automatically using any of the various methods I used to install the package, including using the provided tarball via `install.packages("const.clust_1.3.tar.gz", repo=NULL)` and `install_github`.  I dug a little deeper, with the following results:

I am able to manually load the Fortran module via `dyn.load("path/to/libs/const.clust.so")`.
After doing so, `is.loaded("hcass2")` returns TRUE
If I debug the function `constrained.clust` and run it up to the `.Fortran` call to calculate the necessary arguments to hcass2, I can successfully run it via 
`.Fortran("hcass2", n = as.integer(n), ia = as.integer(hcl$ia), 
         ib = as.integer(hcl$ib), order = integer(n), iia = integer(n), 
         iib = integer(n), PACKAGE ='const.clust')`
It seems like there is a problem with the compiled Fortran library not being included in the package namespace, or not found properly from within that namespace?

I googled, and found [this explanation](http://r.789695.n4.nabble.com/question-re-error-message-package-error-quot-functionName-quot-not-resolved-from-current-namespace-td4663892.html).

**Comment by philippec** [comment]
Thanks Brendan. This repo is currently my personal source for building the packages found at http://numericalecology.com (or, more specifically, http://adn.biol.umontreal.ca/~numericalecology/Rcode/). If there is a need for support, that is where you should go, especially with Dr. Legendre.

**Comment by philippec** [comment]
Just got a reply from Dr. Legendre and not working in 3.2.3 is a known issue due to an external dependency on 'mvpart'. But the installed version (from the above website) should work fine in 3.0.3.

**Comment by brendanf** [comment]
Philippe,

Thanks.  That was actually something I had already resolved (by 
installing mvpart from archived source).  Unfortunately, I'm working on 
Linux, so the installed version isn't an option for me.  On the plus 
side, I have figured out a fix.  If I add
useDynLib(const.clust, hcass2)
to the NAMESPACE file and then rebuild/reinstall the package, everything 
seems to work fine.  I sent a message to Dr. Legendre also today, 
outlining the fix.

Brendan
